src/muya/lib/parser/marked/lexer.js

```diff
@@ -421,4 +425,35 @@ Lexer.prototype.token = function (src, top, bq) {
   return this.tokens
 }
 
+function splitCells (tableRow, count) {
+  // ensure that every cell-delimiting pipe has a space
+  // before it to distinguish it from an escaped pipe
+  let row = tableRow.replace(/\|/g, function (match, offset, str) {
+    let escaped = false
+    let curr = offset
+    while (--curr >= 0 && str[curr] === '\\') escaped = !escaped
+    if (escaped) {
+      // odd number of slashes means | is escaped
+      // so we leave it alone
+      return '|'
+    } else {
+      // add space before unescaped |
+      return ' |'
+    }
+  })
+
+  let cells = row.split(/ \|/)
+  if (cells.length > count) {
+    cells.splice(count)
+  } else {
+    while (cells.length < count) cells.push('')
+  }
+
+  for (let i = 0; i < cells.length; i++) {
+    // leading or trailing whitespace is ignored per the gfm spec
+    cells[i] = cells[i].trim().replace(/\\\|/g, '|')
```

Comment:
> Can you explain the second solution more detail?

@Jocs We just have to remove this line. Markedjs removes the escape character because they have preview mode only.